### DISCUSSION TITLE
What is the proper way to wrap useFetch, useLazyFetch with additional internal async calls

### DISCUSSION THREAD

**Discussion by jdalrymple** [nuxt/nuxt#16711] [Questions]
I've been working on a composable that's wraps the useLazyFetch function, but initially performs an async call to obtain auth credentials prior to firing off the request.

//useAPI.js
```js
export default async (id) => {
  const token = await getToken();

  const { pending, data } = useLazyFetch('myurl', {
     headers: { Authorization: `Bearer ${token}` }
   });

  return {
    pending,
    data,
  };
}
```
However, the way i set it up, its still blocking due to the await call. How can i change the code such that it is not UI blocking?

Here is an example https://github.com/kevinhassan/vue3-vite/blob/6d99a33aa1b79522ce3bd7b508d5a07fa69c3580/src/composables/useApi.js, but in this case, useLazyFetch is not utilized and the statefullness is rewritten in the function itself.

**Comment by sevillaarvin** [comment]
What about using `useLazyAsyncData` and chaining two `$fetch` inside it?

https://v3.nuxtjs.org/getting-started/data-fetching/#uselazyasyncdata

**Comment by jdalrymple** [comment]
Because the fetches are unrelated :(
